dali/operators/geometry/affine_transforms/scale_transform.cc

```diff
@@ -0,0 +1,166 @@
+ */
+class ScaleTransformCPU
+    : public TransformBaseOp<CPUBackend, ScaleTransformCPU> {
+ public:
+  using SupportedDims = dims<1, 2, 3, 4, 5, 6>;
+
+  explicit ScaleTransformCPU(const OpSpec &spec) :
+      TransformBaseOp<CPUBackend, ScaleTransformCPU>(spec),
+      has_scale_input_(spec.HasTensorArgument("scale")),
+      has_center_(spec.HasArgument("center")),
+      has_center_input_(spec.HasTensorArgument("center")) {
+  }
+
+  template <typename T, int mat_dim>
+  void DefineTransforms(span<affine_mat_t<T, mat_dim>> matrices) {
+    constexpr int ndim = mat_dim - 1;
+    assert(matrices.size() == static_cast<int>(scale_.size()));
+    for (int i = 0; i < matrices.size(); i++) {
+      auto &mat = matrices[i];
+      auto *scale = scale_[i].data();
+      mat = affine_mat_t<T, mat_dim>::identity();
+      for (int d = 0; d < ndim; d++) {
+        mat(d, d) = scale[d];
+      }
+
+      if (has_center_ || has_center_input_) {
+        auto *center = center_[i].data();
+        for (int d = 0; d < ndim; d++) {
+          mat(d, ndim) = center[d] * (T(1) - scale[d]);
+        }
+      }
+    }
+  }
+
+  void ProcessScaleConstant(const OpSpec &spec, const workspace_t<CPUBackend> &ws) {
+    scale_.resize(1);
+    scale_[0] = spec.GetArgument<std::vector<float>>("scale");
+    ndim_ = scale_[0].size();
+  }
+
+  void ProcessCenterConstant(const OpSpec &spec, const workspace_t<CPUBackend> &ws) {
+    center_.resize(1);
+    center_[0] = spec.GetArgument<std::vector<float>>("center");
+  }
+
+  void ProcessScaleArgInput(const OpSpec &spec, const workspace_t<CPUBackend> &ws) {
+    const auto& scale = ws.ArgumentInput("scale");
+    auto scale_view = view<const float>(scale);
+    DALI_ENFORCE(is_uniform(scale_view.shape),
+      "All samples in argument ``scale`` should have the same shape");
+    DALI_ENFORCE(scale_view.shape.sample_dim() == 1,
+      "``scale`` must be a 1D tensor");
+
+    scale_.resize(nsamples_);
+    for (int i = 0; i < nsamples_; i++) {
+      scale_[i].resize(ndim_);
+      for (int d = 0; d < ndim_; d++) {
+        scale_[i][d] = scale_view[i].data[d];
+      }
+    }
+  }
+
+  void ProcessCenterArgInput(const OpSpec &spec, const workspace_t<CPUBackend> &ws) {
+    const auto& center = ws.ArgumentInput("center");
+    auto center_view = view<const float>(center);
+    DALI_ENFORCE(is_uniform(center_view.shape),
+      "All samples in argument ``center`` should have the same shape");
+    DALI_ENFORCE(center_view.shape.sample_dim() == 1,
+      "``center`` must be a 1D tensor");
+
+    center_.resize(nsamples_);
+    for (int i = 0; i < nsamples_; i++) {
+      center_[i].resize(ndim_);
+      for (int d = 0; d < ndim_; d++) {
+        center_[i][d] = center_view[i].data[d];
+      }
+    }
+  }
+
+  void ProcessArgs(const OpSpec &spec, const workspace_t<CPUBackend> &ws) {
+    if (has_scale_input_) {
+      ProcessScaleArgInput(spec, ws);
+    } else {
+      ProcessScaleConstant(spec, ws);
+    }
```

Comment:
I guess it could be simplified too.
Something like:
```
struct InputArg {
  bool has;
  bool has_input;
  string name;
}
InputArg scale_arg;
InputArg centre_arg;

Process(scale_arg);
Process(centre_arg);
```